pkg/discovery/dtofactory/application_entity_dto_builder.go

```diff
@@ -43,130 +42,184 @@ func NewApplicationEntityDTOBuilder(sink *metrics.EntityMetricSink) *application
 	}
 }
 
-// Build entityDTOs based on the given pod list.
+// get hosting node cpu frequency
+func (builder *applicationEntityDTOBuilder) getNodeCPUFrequency(pod *api.Pod) (float64, error) {
+	key := util.NodeKeyFromPodFunc(pod)
+	cpuFrequencyUID := metrics.GenerateEntityStateMetricUID(task.NodeType, key, metrics.CpuFrequency)
+	cpuFrequencyMetric, err := builder.metricsSink.GetMetric(cpuFrequencyUID)
+	if err != nil {
+		err := fmt.Errorf("Failed to get cpu frequency from sink for node %s: %v", key, err)
+		glog.Error(err)
+		return 0.0, err
+	}
+
+	cpuFrequency := cpuFrequencyMetric.GetValue().(float64)
+	return cpuFrequency, nil
+}
+
 func (builder *applicationEntityDTOBuilder) BuildEntityDTOs(pods []*api.Pod) ([]*proto.EntityDTO, error) {
 	var result []*proto.EntityDTO
-	for _, pod := range pods {
-		// id. Application entity dto ID is consisted of appPrefix and pod UID.
-		appID := getApplicationID(string(pod.UID))
-		entityDTOBuilder := sdkbuilder.NewEntityDTOBuilder(proto.EntityDTO_APPLICATION, appID)
-
-		// display name
-		displayName := AppPrefix + util.GetPodClusterID(pod)
-		entityDTOBuilder.DisplayName(displayName)
 
-		// commodity sold
-		commoditiesSold, err := builder.getApplicationCommoditiesSold(pod)
+	for _, pod := range pods {
+		nodeCPUFrequency, err := builder.getNodeCPUFrequency(pod)
 		if err != nil {
-			glog.Errorf("Failed to create commodities sold by application %s: %s", displayName, err)
+			glog.Errorf("failed to build ContainerDTOs for pod[%s]: %v", pod.Name, err)
```

Comment:
Here I suggest to print out not only `pod.Name`, but also `pod.Namespace`. So maybe use `util.GetPodClusterID(pod)`.